### PR TITLE
metrics and traces support

### PR DESCRIPTION
**Pull request by rockb1017**
## Proposed changes

it now support collecting and sending metrics to both splunk and o11y, and sending traces to o11y. 
it has partial gateway logic implemented and it is to be completed later.

### CODE REVIEW

File: charts/sck-otel/ci/filled-values.yaml [hunk 1]
```diff
@@ -243,32 +246,26 @@ agent:
...
       protocol: TCP
       enabled:  # true for splunkObservability and if tracesEnabled=true
     jaeger-grpc:
       containerPort: 14250
-      hostPort: 14250
       protocol: TCP
       enabled:  # true for splunkObservability and if tracesEnabled=true
     signalfx:
       containerPort: 9943
-      hostPort: 9943
```
Comment (dmitryax) [inline charts/sck-otel/ci/filled-values.yaml:271]: Why removed? host ports are required in the agent mode to receive traces from local pods

File: charts/sck-otel/ci/o11y-values.yaml [hunk 2]
```diff
@@ -14,3 +14,29 @@ splunkObservability:
...
+      protocol: TCP
+      enabled: true
+    jaeger-grpc:
+      containerPort: 14250
+      protocol: TCP
+      enabled: true
+    signalfx:
+      containerPort: 9943
+      protocol: TCP
+      enabled: true
```
Comment (dmitryax) [inline charts/sck-otel/ci/o11y-values.yaml:42]: why explicitly add them here? it looks the same as default

File: charts/sck-otel/ci/splunk-values.yaml [hunk 3]
```diff
@@ -45,3 +45,29 @@ containerLogs:
...
+      protocol: TCP
+      enabled: true
+    jaeger-grpc:
+      containerPort: 14250
+      protocol: TCP
+      enabled: true
+    signalfx:
+      containerPort: 9943
+      protocol: TCP
+      enabled: true
```
Comment (dmitryax) [inline charts/sck-otel/ci/splunk-values.yaml:73]: same q here

File: charts/sck-otel/templates/_config-Agent.tpl [hunk 4]
```diff
@@ -359,13 +455,78 @@ service:
...
+        {{- if eq (include "splunk-otel-collector.sendMetricsToO11y" .) "true" }}
+        - signalfx
+        {{- end }}
+        {{- if eq (include "splunk-otel-collector.sendMetricsToSplunk" .) "true" }}
+        - splunk_hec/platformMetrics
+        {{- end }}
+        {{- end }}
+    {{- end }}
+
+    {{- if eq (include "splunk-otel-collector.collectMetric" .) "true" }}
```
Comment (dmitryax) [inline charts/sck-otel/templates/_config-Agent.tpl]: These internal otel-collector metrics should be always sent to o11y if it's enabled. no matter which telemetry is enabled. so the condition here should be `splunk-otel-collector.sendMetricsToSplunk or splunk-otel-collector.splunkO11yEnabled`.

File: charts/sck-otel/templates/_config-cluster-data-collector.tpl [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    metrics/collector:
+      receivers: [prometheus/k8s_cluster_receiver]
+      processors:
+        - memory_limiter
+        - batch
+        - resource
+        - resource/add_collector_k8s
+        - resourcedetection
+      exporters: 
+        {{- if eq (include "splunk-otel-collector.sendMetricsToO11y" .) "true" }}
```
Comment (dmitryax) [inline charts/sck-otel/templates/_config-cluster-data-collector.tpl]: Same comment here

**Review by dmitryax — COMMENTED** [review]
Thanks for putting this together! Overall looks very good. I'll make another pass after this one

File: charts/sck-otel/ci/o11y-values.yaml (hunk 2 above)
Comment (rockb1017) [reply charts/sck-otel/ci/o11y-values.yaml:42]: i removed hostPort. because it says no available port on host in the ci env.

File: charts/sck-otel/ci/filled-values.yaml (hunk 1 above)
Comment (rockb1017) [reply charts/sck-otel/ci/filled-values.yaml:271]: this is only for chart testing in github actions

File: charts/sck-otel/templates/_config-Agent.tpl (hunk 4 above)
Comment (dmitryax) [reply charts/sck-otel/templates/_config-Agent.tpl]: Doesn't look like this is resolved. the internal collector metrics should be always sent to o11y if `otel-collector.splunkO11yEnabled`, even with `.Values.splunkObservability.metricsEnabled=false`

File: charts/sck-otel/templates/_pod.tpl [hunk 6]
```diff
@@ -31,18 +31,49 @@ containers:
...
+            fieldPath: metadata.name
+      - name: K8S_POD_UID
+        valueFrom:
+          fieldRef:
+            fieldPath: metadata.uid
+      - name: K8S_NAMESPACE
+        valueFrom:
+          fieldRef:
+            fieldPath: metadata.namespace
+      {{- if .Values.metricsEnabled }}
```
Comment (dmitryax) [inline charts/sck-otel/templates/_pod.tpl]: We don't have this values. Should be `splunk-otel-collector.collectMetric` I believe

**Review by dmitryax — COMMENTED** [review]
Looks good overall! just couple of nits

File: charts/sck-otel/templates/_pod.tpl [hunk 7]
```diff
@@ -6,6 +6,21 @@ imagePullSecrets:
...
+    command: ['sh', '-c', "chown -Rv 20000:20000 {{ .Values.checkpointPath }}; chmod -v g+rwx {{ .Values.checkpointPath }}{{ if .Values.containerLogs.enabled }}; setfacl -R -d -m g:20000:rx /var/log;setfacl -R -d -m g:20000:rx /var/lib/docker/containers{{ end }}"]
+    volumeMounts:
+      - name: checkpoint
+        mountPath: {{ .Values.checkpointPath }}    
+      {{- if .Values.containerLogs.enabled }}
+      - name: varlog
+        mountPath: /var/log
+      - name: varlibdockercontainers
+        mountPath: /var/lib/docker/containers
+      {{- end }}
```
Comment (dmitryax) [inline charts/sck-otel/templates/_pod.tpl]: Is this expected addition? Looks unrelated

Comment (rockb1017) [reply charts/sck-otel/templates/_pod.tpl]: oh shoot, i was playing with it and i shouldn't have included it. let me revert it.

**Comment by dmitryax** [comment]
@rockb1017 looks like you reverted too much. `{{- if .Values.metricsEnabled }}` part came back :)

**Review by dmitryax — APPROVED** [review]
LGTM
